src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/SignatureDescription.cs

```diff
@@ -27,21 +29,23 @@ public SignatureDescription(SecurityElement el)
 
         public virtual AsymmetricSignatureDeformatter CreateDeformatter(AsymmetricAlgorithm key)
         {
-            AsymmetricSignatureDeformatter item = (AsymmetricSignatureDeformatter)CryptoConfig.CreateFromName(DeformatterAlgorithm);
-            item.SetKey(key);
+            AsymmetricSignatureDeformatter? item = (AsymmetricSignatureDeformatter?)CryptoConfig.CreateFromName(DeformatterAlgorithm!);
+            item!.SetKey(key);
             return item;
         }
 
         public virtual AsymmetricSignatureFormatter CreateFormatter(AsymmetricAlgorithm key)
         {
-            AsymmetricSignatureFormatter item = (AsymmetricSignatureFormatter)CryptoConfig.CreateFromName(FormatterAlgorithm);
+            AsymmetricSignatureFormatter? item = (AsymmetricSignatureFormatter?)CryptoConfig.CreateFromName(FormatterAlgorithm!);
+            Debug.Assert(item != null);
             item.SetKey(key);
             return item;
         }
 
-        public virtual HashAlgorithm CreateDigest()
+        public virtual HashAlgorithm? CreateDigest()
         {
-            return (HashAlgorithm)CryptoConfig.CreateFromName(DigestAlgorithm);
+            Debug.Assert(DigestAlgorithm != null);
```

Comment:
This isn't a valid assert (it's a get/set property that doesn't restrict prevent null)